packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -56,16 +57,23 @@ function UnconnectedNavigatorProvider(
 	props: WordPressComponentProps< NavigatorProviderProps, 'div' >,
 	forwardedRef: ForwardedRef< any >
 ) {
-	const { initialPath, children, className, ...otherProps } =
-		useContextSystem( props, 'NavigatorProvider' );
+	const {
+		initialPath,
+		location: locationProp,
+		onChange,
+		children,
+		className,
+		...otherProps
+	} = useContextSystem( props, 'NavigatorProvider' );
+	const [ location, updateLocation ] = useControlledValue( {
+		onChange,
+		value: locationProp,
+		defaultValue: { path: initialPath },
+	} );
 
 	const [ locationHistory, setLocationHistory ] = useState<
 		NavigatorLocation[]
-	>( [
-		{
-			path: initialPath,
-		},
-	] );
+	>( [ location ?? { path: initialPath } ] );
```

Comment:
Given that `useControlledValue` is already handling the fallback to `{ path: initialPath }` as the default value when `locationProp` is not defined, I believe that this fallback is extra and can be removed

```suggestion
	>( location ? [ location ] : [] );
```